Cubical/Data/Rationals/Order.agda

```diff
@@ -0,0 +1,626 @@
+                → ((a ℤ.· ℕ₊₁→ℤ d) ℤ.< (c ℤ.· ℕ₊₁→ℤ b))
+                ≡ ((a ℤ.· ℕ₊₁→ℤ f) ℤ.< (e ℤ.· ℕ₊₁→ℤ b))
+        lemma< (a , b) (c , d) (e , f) cf≡ed = (ua (propBiimpl→Equiv ℤ.isProp< ℤ.isProp<
+                (ℤ.<-·o-cancel {k = -1+ d} ∘
+                  subst2 ℤ._<_ (·CommR a (ℕ₊₁→ℤ d) (ℕ₊₁→ℤ f))
+                               (·CommR c (ℕ₊₁→ℤ b) (ℕ₊₁→ℤ f) ∙
+                                cong (ℤ._· ℕ₊₁→ℤ b) cf≡ed ∙
+                                ·CommR e (ℕ₊₁→ℤ d) (ℕ₊₁→ℤ b)) ∘
+                 ℤ.<-·o {k = -1+ f})
+                (ℤ.<-·o-cancel {k = -1+ f} ∘
+                  subst2 ℤ._<_ (·CommR a (ℕ₊₁→ℤ f) (ℕ₊₁→ℤ d))
+                               (·CommR e (ℕ₊₁→ℤ b) (ℕ₊₁→ℤ d) ∙
+                                cong (ℤ._· ℕ₊₁→ℤ b) (sym cf≡ed) ∙
+                                ·CommR c (ℕ₊₁→ℤ f) (ℕ₊₁→ℤ b)) ∘
+                 ℤ.<-·o {k = -1+ d})))
+
+        fun₀ : ℤ × ℕ₊₁ → ℚ → hProp ℓ-zero
+        fst (fun₀ (a , b) [ c , d ]) = a ℤ.· ℕ₊₁→ℤ d ℤ.< c ℤ.· ℕ₊₁→ℤ b
+        snd (fun₀ _ [ _ ]) = ℤ.isProp<
+        fun₀ a/b (eq/ c/d e/f cf≡ed i) = record
+          { fst = lemma< a/b c/d e/f cf≡ed i
+          ; snd = isProp→PathP (λ i → isPropIsProp {A = lemma< a/b c/d e/f cf≡ed i}) ℤ.isProp< ℤ.isProp< i
+          }
+        fun₀ a/b (squash/ x y p q i j) = isSet→SquareP (λ _ _ → isSetHProp)
+          (λ _ → fun₀ a/b x)
+          (λ _ → fun₀ a/b y)
+          (λ i → fun₀ a/b (p i))
+          (λ i → fun₀ a/b (q i)) j i
+
+        toPath : ∀ a/b c/d (x : a/b ∼ c/d) (y : ℚ) → fun₀ a/b y ≡ fun₀ c/d y
+        toPath (a , b) (c , d) ad≡cb = elimProp (λ _ → isSetHProp _ _) λ (e , f) →
+          Σ≡Prop (λ _ → isPropIsProp) (ua (propBiimpl→Equiv ℤ.isProp< ℤ.isProp<
+                (ℤ.<-·o-cancel {k = -1+ b} ∘
+                  subst2 ℤ._<_ (·CommR a (ℕ₊₁→ℤ f) (ℕ₊₁→ℤ d) ∙
+                                cong (ℤ._· ℕ₊₁→ℤ f) ad≡cb ∙
+                                ·CommR c (ℕ₊₁→ℤ b) (ℕ₊₁→ℤ f))
+                               (·CommR e (ℕ₊₁→ℤ b) (ℕ₊₁→ℤ d)) ∘
+                 ℤ.<-·o {k = -1+ d})
+                (ℤ.<-·o-cancel {k = -1+ d} ∘
+                  subst2 ℤ._<_ (·CommR c (ℕ₊₁→ℤ f) (ℕ₊₁→ℤ b) ∙
+                                cong (ℤ._· ℕ₊₁→ℤ f) (sym ad≡cb) ∙
+                                ·CommR a (ℕ₊₁→ℤ d) (ℕ₊₁→ℤ f))
+                               (·CommR e (ℕ₊₁→ℤ d) (ℕ₊₁→ℤ b)) ∘
+                 ℤ.<-·o {k = -1+ b})))
+
+        fun : ℚ → ℚ → hProp ℓ-zero
+        fun [ a/b ] y = fun₀ a/b y
+        fun (eq/ a/b c/d ad≡cb i) y = toPath a/b c/d ad≡cb y i
+        fun (squash/ x y p q i j) z = isSet→SquareP (λ _ _ → isSetHProp)
+          (λ _ → fun x z) (λ _ → fun y z) (λ i → fun (p i) z) (λ i → fun (q i) z) j i
+
+_≤_ : ℚ → ℚ → Type₀
+m ≤ n = fst (m ≤' n)
+
+_<_ : ℚ → ℚ → Type₀
+m < n = fst (m <' n)
+
+_≥_ : ℚ → ℚ → Type₀
+m ≥ n = n ≤ m
+
+_>_ : ℚ → ℚ → Type₀
+m > n = n < m
+
+_#_ : ℚ → ℚ → Type₀
+m # n = (m < n) ⊎ (n < m)
+
+data Trichotomy (m n : ℚ) : Type₀ where
```

Comment:
We do, but the issue here is that it only works for [ a , b ] [ c , d ] cases. Upstreaming it would require a gnarly pattern match like the definition of < or <=/